File: Easy/2025-10-28-3354-Make-Array-Elements-Equal-to-Zero/solution.java
class Solution {
    public int countValidSelections(int[] nums) {
        int n = nums.length;
        int validCount = 0;

        for (int i = 0; i < n; i++) {
            if (nums[i] == 0) {
                if (canMakeAllZero(nums, i, 1) || canMakeAllZero(nums, i, -1)) {
                    validCount++;
                }
            }
        }

        return validCount;
    }

    private boolean canMakeAllZero(int[] nums, int start, int direction) {
        int[] copy = nums.clone();
        int curr = start;

        while (curr >= 0 && curr < copy.length) {
            if (copy[curr] == 0) {
                curr += direction;
            } else {
                copy[curr]--;
                direction = -direction; // reverse direction
                curr += direction;
            }
        }

        for (int num : copy) {
            if (num != 0) {
                return false;
            }
        }
        return true;
    }
}

File: Easy/2025-10-28-3354-Make-Array-Elements-Equal-to-Zero/solution.js
var countValidSelections = function(nums) {
    const n = nums.length;
    let validCount = 0;

    for (let start = 0; start < n; start++) {
        if (nums[start] === 0) {
            for (let direction = -1; direction <= 1; direction += 2) {
                let curr = start;
                const tempNums = [...nums];
                let valid = true;

                while (valid && curr >= 0 && curr < n) {
                    if (tempNums[curr] === 0) {
                        curr += direction;
                    } else {
                        tempNums[curr]--;
                        direction *= -1; // reverse direction
                        curr += direction;
                    }
                }

                if (valid && curr < 0 || curr >= n) {
                    validCount++;
                }
            }
        }
    }

    return validCount;
};